check.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+#
+# LICENSE: Public Domain
+
+
+echo "CHECK that All translator have a translatorID"
+haveid=$(grep -c '"translatorID"' *.js | awk -F: '$2!="1" {print $1,"found",$2,"translatorID"}')
+if [[ -n "$haveid" ]];then
+  echo "Error: no translatorID or multiple translatorIDs"
+  echo "$haveid"
+  exit 1
+fi
+
+echo "CHECK that all translatorIDs are unique"
+noduplicate=$( { (grep '"translatorID"' *.js | awk -F: '{print $3}') & (awk -F# 'NR>2 { print $1 }' deleted.txt) } | sed 's/[", ]//g' | sort | uniq -cd)
+echo "$noduplicate"
+if [[ -n "$noduplicate" ]];then
+  echo "Error: duplicate translatorID found"
+  echo "$noduplicate"
+  exit 1
+fi
+
+echo "CHECK that deleted translator are added to deleted.txt and number is increased"
+deleted=$(git diff --cached --name-only --diff-filter=D -- '*.js*')
+if [[ -n "$deleted" ]];then
+  # number increased
+  oldrevision=$(git show HEAD:deleted.txt | awk 'NR<2 { print $1 }')
```

Comment:
I think your comparisons are all off here. Doesn't Travis just get a checkout of the repo after the new commit? So HEAD is just the current state of the file.